controllers/ciresource_fsm.go

```diff
@@ -167,7 +169,7 @@ func (f *CIResourceFSM) handleStateInUse(context CIResourceFSMContext) (time.Dur
 		return f.TriggerEvent("released")
 	case ofcirv1.StateInUse:
 		// A fallback resource has been requested, so it must be provisioned
-		if context.CIPool.IsFallbackPool() && context.CIResource.Status.Address == "" {
+		if context.CIPool.IsFallbackPool() && context.CIResource.Status.ResourceId == fallbackResourceID {
```

Comment:
nit: probablly leaving also the check on the address won't hurt

Reply:
added the address check also